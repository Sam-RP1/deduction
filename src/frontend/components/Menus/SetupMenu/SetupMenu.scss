@use '../../../styles/base';
@import '../../../styles/partials/color-themes';
@import '../../../styles/mixins/transition';

%flex-center-column {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.setup-menu {
    @extend %flex-center-column;
    height: 100%;
    width: 100%;
    @include themed() {
        color: t($text);
    }

    &__option {
        display: flex;
        flex-direction: column;
        height: auto;
        width: 100%;
        max-width: 600px;
        margin-bottom: 25px;

        h3 {
            font-size: 1.9em;
            font-weight: 400;
        }

        h3,
        p {
            margin: 0;
            @include transition-theme(color 0.4s ease);
        }

        &__row {
            display: flex;
            flex-direction: row;
            align-items: center;
            justify-content: space-between;
            margin-bottom: 5px;
        }

        &__selection {
            @extend .setup-menu__option;
            h3 {
                margin-bottom: 6px;
            }

            &__container {
                display: flex;
                flex-flow: row wrap;
            }

            &__brick {
                @extend %flex-center-column;
                cursor: pointer;
                position: relative;
                height: 50px;
                min-width: 90px;
                width: auto;
                margin-right: 10px;
                box-sizing: border-box;
                padding: 5px;
                border: 3px solid;
                @include themed() {
                    border-color: t($border);
                }
                @include transition-theme(border-color 0.4s ease, background-color 0.3s ease);

                &[aria-checked='true'] {
                    @include themed() {
                        color: t($text-secondary);
                        background-color: base.$selected;
                        &:hover {
                            background-color: base.$remove;
                        }
                    }
                }

                p {
                    text-transform: uppercase;
                }

                &:hover {
                    @include themed() {
                        color: t($text-secondary);
                        background-color: t($selected);
                    }
                }
            }
        }

        &__text-input {
            @extend .setup-menu__option;
            h3 {
                margin-bottom: 4px;
                &:first-child {
                    margin-bottom: 20px;
                }
            }

            input {
                font-size: 1.1em;
                margin-top: 5px;
                margin-bottom: 5px;
            }

            &__bricks {
                display: flex;
                flex-flow: row wrap;

                div {
                    cursor: pointer;
                    position: relative;
                    height: auto;
                    width: auto;
                    margin: 5px 10px 0 0;
                    box-sizing: border-box;
                    padding: 5px;
                    border: 3px solid;
                    @include themed() {
                        border-color: t($border);
                    }
                    @include transition-theme(border-color 0.4s ease, background-color 0.3s ease);

                    p {
                        text-transform: uppercase;
                    }

                    &:hover {
                        @include themed() {
                            color: t($text-secondary);
                        }
                        background-color: base.$remove;
                    }
                }
            }
        }
    }
}
