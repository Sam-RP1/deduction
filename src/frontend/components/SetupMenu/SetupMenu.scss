@import '../../styles/partials/color-themes';
@import '../../styles/mixins/transition';

%flex-center-column {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.setup-menu {
  @extend %flex-center-column;
  min-height: calc(100vh - 32px - 80px);
  height: auto;
  width: 100%;
  @include themed() {
    color: t($text);
  }
  @include transition-theme(color 0.4s ease, border-color 0.4s ease);

  &__title {
    margin: 0;
    font-size: 5em;
    font-weight: 400;
    margin-bottom: 25px;
  }

  &__option {
    display: flex;
    flex-direction: column;
    height: auto;
    width: 100%;
    max-width: 900px;
    margin-bottom: 25px;

    h3, p {
      margin: 0;
    }

    h3 {
      font-size: 1.9em;
      font-weight: 400;
    }

    &__toggle {
      &__row {
        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: space-between;
        margin-bottom: 5px;
      }
    }

    &__word-select {
      h3 {
        margin-bottom: 6px;
      }

      &__container {
        display: flex;
        flex-flow: row wrap;
      }

      &__word-brick {
        @extend %flex-center-column;
        cursor: pointer;
        position: relative;
        min-height: 40px;
        height: auto;
        max-height: 60px;
        min-width: 90px;
        width: auto;
        max-width: 100%;
        margin-right: 10px;
        box-sizing: border-box;
        padding: 5px;
        border: 3px solid;
        @include themed() {
          border-color: t($border);
        }
        @include transition-theme(border-color 0.4s ease, background-color 0.3s ease);

        p {
          text-transform: uppercase;
        }

        &:hover {
          @include themed() {
            color: t($text-secondary);
            background-color: t($selected);
          }
        }
      }
    }

    &__word-entry {
      h3:first-child {
        margin-bottom: 20px;
      }

      h3 {
        margin-bottom: 4px;
      }
    }
  }
}
