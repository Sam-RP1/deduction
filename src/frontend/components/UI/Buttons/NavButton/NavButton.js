import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import '../Button.scss';

const NavButton = (props) => {
    return (
        <NavLink
            to={props.route}
            exact={true}
            onClick={() => {
                props.function();
            }}
        >
            <button className={'btn ' + props.opClasses}>{props.title}</button>
        </NavLink>
    );
};

NavButton.propTypes = {
    title: PropTypes.string.isRequired,
    route: PropTypes.string.isRequired,
    function: PropTypes.func.isRequired,
    opClasses: PropTypes.string,
};

NavButton.defaultProps = {
    title: 'UNDEFINED',
    route: '/undefined',
    function: () => console.log('[BUTTON] undefined button function'),
    opClasses: '',
};

export default NavButton;
