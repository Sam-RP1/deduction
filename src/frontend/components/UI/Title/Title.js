import React from 'react';
import PropTypes from 'prop-types';

import './Title.scss';

const Title = (props) => {
    return <h1 className='title'>{props.title}</h1>;
};

Title.propTypes = {
    title: PropTypes.string,
};

Title.defaultProps = {
    title: 'Deduction',
};

export default Title;
