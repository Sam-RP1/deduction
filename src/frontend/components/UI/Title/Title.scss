@import '../../../styles/partials/color-themes';
@import '../../../styles/mixins/transition';

.title {
  height: 90px;
  margin: 0;
  font-size: 5em;
  font-weight: 400;
  @include themed() { color: t($text); }
  @include transition-theme(color 0.4s ease);
  @media screen and (max-width: 350px) { font-size: 4.5em; }
}
