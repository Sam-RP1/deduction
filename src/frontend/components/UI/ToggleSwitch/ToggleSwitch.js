import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import './ToggleSwitch.scss';

const ToggleSwitch = (props) => {
    const [isToggled, setToggleState] = useState(false);

    const toggle = () => {
        isToggled ? setToggleState(false) : setToggleState(true);
    };

    useEffect(() => {
        if (props.toggle) {
            setToggleState(true);
        }
    }, []);

    return (
        <div
            onClick={() => {
                props.function();
                toggle();
            }}
            className={'toggle-switch ' + (isToggled ? 'active' : '')}
        >
            <span></span>
        </div>
    );
};

ToggleSwitch.propTypes = {
    toggle: PropTypes.bool,
    function: PropTypes.func.isRequired,
};

ToggleSwitch.defaultProps = {
    toggle: false,
    function: () => console.log('[TOGGLE SWITCH] undefined toggle switch function'),
};

export default ToggleSwitch;
