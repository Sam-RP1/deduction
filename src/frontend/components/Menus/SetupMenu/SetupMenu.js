import React from 'react';
import PropTypes from 'prop-types';

// Imports
import Title from '../../UI/Title/Title';
import ToggleSwitch from '../../UI/ToggleSwitch/ToggleSwitch';
import NavButton from '../../UI/Buttons/NavButton/NavButton';

// Styles
import './SetupMenu.scss';

// Presentational Component
const SetupMenu = (props) => {
    return (
        <section className='setup-menu'>
            <Title title='Deduction' />

            <div className='setup-menu__option'>
                <div className='setup-menu__option__row'>
                    <h3>Turn Timer:</h3>
                    <ToggleSwitch function={props.toggleTurnTimer} />
                </div>
                <p>If enabled teams have one minute to complete their turn.</p>
            </div>

            <div className='setup-menu__option'>
                <div className='setup-menu__option__row'>
                    <h3>Quick Game:</h3>
                    <ToggleSwitch function={props.toggleQuickGame} />
                </div>
                <p>
                    If enabled games will last upto six minutes before ending & teams have 30 seconds to complete each
                    turn.
                </p>
            </div>

            <div className='setup-menu__option__selection'>
                <h3>Select a Word Group...</h3>
                <div className='setup-menu__option__selection__container'>
                    {props.wordGroups.map((item) => {
                        return (
                            <div
                                key={item.id}
                                id={item.id}
                                role='checkbox'
                                aria-checked='false'
                                onClick={() => {
                                    props.wordGroupHandler(item.id);
                                }}
                                className='setup-menu__option__selection__brick'
                            >
                                <p>{item.title}</p>
                            </div>
                        );
                    })}
                </div>
            </div>

            <div className='setup-menu__option__text-input'>
                <h3>OR</h3>
                <h3>Enter Custom Words...</h3>
                <p>Enter 25 words with a , (comma) seperating each word.</p>
                <textarea onChange={props.customWordsHandler} />
            </div>

            <NavButton title={'Go!'} route={'/game'} />
        </section>
    );
};

SetupMenu.propTypes = {
    toggleTurnTimer: PropTypes.func.isRequired,
    toggleQuickGame: PropTypes.func.isRequired,
    wordGroups: PropTypes.array.isRequired,
    wordGroupHandler: PropTypes.func.isRequired,
    customWordsHandler: PropTypes.func.isRequired,
    submitHandler: PropTypes.func.isRequired,
};

SetupMenu.defaultProps = {
    toggleTurnTimer: () => {
        console.log('[TURN TIMER] toggleTurnTimer err');
    },
    toggleQuickGame: () => {
        console.log('[QUICK GAME] toggleQuickGame err');
    },
    wordGroups: [{ id: 'err', title: 'ERROR :(' }],
    wordGroupHandler: () => {
        console.log('[WORD GROUP] wordGroupHandler err');
    },
    customWordsHandler: () => {
        console.log('[CUSTOM WORDS] customWordsHandler err');
    },
    submitHandler: () => {
        console.log('[SUBMIT BTN] submitHandler err');
    },
};

export default SetupMenu;
