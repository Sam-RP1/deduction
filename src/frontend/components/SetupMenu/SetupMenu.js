import React, { useState } from 'react';

import ToggleSwitch from '../UI/ToggleSwitch/ToggleSwitch';
import NavButton from '../UI/Buttons/NavButton/NavButton';

import '../../styles/root.scss';
import './SetupMenu.scss';

const SetupMenu = () => {
  const [selectedWordGroup, setSelectedWordGroup] = useState(null);

  const wordGroups = [
    { id: "eng-standard", title: "english" },
    { id: "tbd", title: "tbd" }
  ];

  const wordGroupHandler = (id) => {
    if (id !== selectedWordGroup && selectedWordGroup !== null) {
      const prevSelected = document.getElementById(selectedWordGroup);
      const curSelected = document.getElementById(id);
      prevSelected.setAttribute('aria-checked', "false");
      curSelected.setAttribute('aria-checked', "true");
      setSelectedWordGroup(id);
    } else if (id !== selectedWordGroup && selectedWordGroup === null) {
      const curSelected = document.getElementById(id);
      curSelected.setAttribute('aria-checked', "true");
      setSelectedWordGroup(id);
    }
  }

  return (
    <section className="setup-menu">
    <h1 className="setup-menu__title">Deduction</h1>

    <div className="setup-menu__option setup-menu__option__toggle">
    <div className="setup-menu__option__toggle__row">
    <h3>Turn Timer:</h3>
    <ToggleSwitch function={() => console.log("[TURN TIMER] placeholder function")} />
    </div>
    <p>If enabled teams have one minute to complete their turn.</p>
    </div>

    <div className="setup-menu__option setup-menu__option__toggle">
    <div className="setup-menu__option__toggle__row">
    <h3>Quick Game:</h3>
    <ToggleSwitch function={() => console.log("[TURN TIMER] placeholder function")} />
    </div>
    <p>If enabled games will last upto six minutes before ending & teams have 30 seconds to complete each turn.</p>
    </div>

    <div className="setup-menu__option setup-menu__option__word-select">
    <h3>Select a Word Group...</h3>
    <div className="setup-menu__option__word-select__container">
    {wordGroups.map(item => {
      return (
        <div key={item.id} id={item.id} role="checkbox" aria-checked="false" onClick={() => {wordGroupHandler(item.id);}} className="setup-menu__option__word-select__word-brick">
        <p>{item.title}</p>
        </div>
      )
    })}
    </div>
    </div>

    <div className="setup-menu__option setup-menu__option__word-entry">
    <h3>OR</h3>
    <h3>Enter Custom Words...</h3>
    <p>Enter 25 words with a &apos;,&apos; (comma) seperating each word.</p>
    <textarea/>
    </div>

    <NavButton title={"Go!"} route={"/game"} />

    </section>
  )
}

export default SetupMenu;
