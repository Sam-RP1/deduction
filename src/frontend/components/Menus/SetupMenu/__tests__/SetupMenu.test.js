import React from 'react';

// Enzyme Imports
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

// Component Imports
import SetupMenu from '../SetupMenu';
import Title from '../../../UI/Title/Title';
import ToggleSwitch from '../../../UI/ToggleSwitch/ToggleSwitch';
import NavButton from '../../../UI/Buttons/NavButton/NavButton';

// Enzyme configuration
configure({ adapter: new Adapter() });

// Test cases
describe('<SetupMenu /> | Component', () => {
    const dummyData = [
        { id: 'eng-standard', title: 'english' },
        { id: 'test', title: 'test' },
    ];

    it('should render <Title />', () => {
        const wrapper = shallow(<SetupMenu />);
        expect(wrapper.find(<Title />));
    });

    it('should render <ToggleSwitch />', () => {
        const wrapper = shallow(<SetupMenu />);
        expect(wrapper.find(<ToggleSwitch />));
    });

    it('should render <NavButton />', () => {
        const wrapper = shallow(<SetupMenu />);
        expect(wrapper.find(<NavButton />));
    });

    it('should render <Title />, <ToggleSwitch /> and <NavButton />', () => {
        const wrapper = shallow(<SetupMenu />);
        expect(wrapper.find(<Title />));
        expect(wrapper.find(<ToggleSwitch />));
        expect(wrapper.find(<NavButton />));
    });

    it('should render one selection brick', () => {
        const wrapper = shallow(<SetupMenu wordGroups={[dummyData[0]]} />);
        expect(wrapper.find('.setup-menu__option__selection__container').children().length).toEqual(1);
    });

    it('should render two selection bricks', () => {
        const wrapper = shallow(<SetupMenu wordGroups={dummyData} />);
        expect(wrapper.find('.setup-menu__option__selection__container').children().length).toEqual(2);
    });

    it('should render correctly', () => {
        const wrapper = shallow(<SetupMenu />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
