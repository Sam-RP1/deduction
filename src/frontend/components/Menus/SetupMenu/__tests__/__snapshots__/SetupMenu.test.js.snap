// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SetupMenu /> | Component should render correctly 1`] = `
<section
  className="setup-menu"
>
  <Title
    title="Deduction"
  />
  <div
    className="setup-menu__option"
  >
    <div
      className="setup-menu__option__row"
    >
      <h3>
        Turn Timer:
      </h3>
      <ToggleSwitch
        function={[Function]}
      />
    </div>
    <p>
      If enabled teams have one minute to complete their turn.
    </p>
  </div>
  <div
    className="setup-menu__option"
  >
    <div
      className="setup-menu__option__row"
    >
      <h3>
        Quick Game:
      </h3>
      <ToggleSwitch
        function={[Function]}
      />
    </div>
    <p>
      If enabled games will last upto six minutes before ending & teams have 30 seconds to complete each turn.
    </p>
  </div>
  <div
    className="setup-menu__option__selection"
  >
    <h3>
      Select a Word Group...
    </h3>
    <div
      className="setup-menu__option__selection__container"
    >
      <div
        aria-checked="false"
        className="setup-menu__option__selection__brick"
        id="err"
        key="err"
        onClick={[Function]}
        role="checkbox"
      >
        <p>
          ERROR :(
        </p>
      </div>
    </div>
  </div>
  <div
    className="setup-menu__option__text-input"
  >
    <h3>
      OR
    </h3>
    <h3>
      Enter Custom Words...
    </h3>
    <p>
      Enter 25 words with a , (comma) seperating each word.
    </p>
    <textarea />
  </div>
  <NavButton
    function={[Function]}
    opClasses=""
    route="/game"
    title="Go!"
  />
</section>
`;
